jax/_src/lax/slicing.py

```diff
@@ -490,6 +491,63 @@ def scatter_max(
       indices_are_sorted=indices_are_sorted, unique_indices=unique_indices,
       mode=GatherScatterMode.from_any(mode))
 
+# To avoid recompilation, we store a dict of weak references to funcs.
+_scatter_apply_cache: weakref.WeakKeyDictionary = weakref.WeakKeyDictionary()
+
+def scatter_apply(
```

Comment:
One catch with this is that scatter_p is not differentiable...

Document that, here and below?

Reply:
Ah, thanks, will do. Maybe eventually a dedicated `scatter_apply` primitive would be useful then.